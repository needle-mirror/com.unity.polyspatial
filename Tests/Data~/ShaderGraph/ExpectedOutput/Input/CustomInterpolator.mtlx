<?xml version ="1.0"?>
<materialx version = "1.38" colorspace = "lin_rec709">
    <geomcolor name = "_1" type = "color4">
    </geomcolor>
    <texcoord name = "_2" type = "vector2">
        <input name = "index" type = "integer" value = "0" />
    </texcoord>
    <UsdPrimvarReader name = "_3" type = "vector2">
        <input name = "varname" type = "string" value = "vertexUV1" />
    </UsdPrimvarReader>
    <realitykit_surface_custom_attribute name = "_4" type = "vector4">
    </realitykit_surface_custom_attribute>
    <convert name = "_5" type = "vector4">
        <input name = "in" type = "vector3" nodename = "_6" />
    </convert>
    <multiply name = "_6" type = "vector3">
        <input name = "in1" type = "vector3" nodename = "_7" />
        <input name = "in2" type = "vector3" value = "1,1,-1" />
    </multiply>
    <position name = "_7" type = "vector3">
        <input name = "space" type = "string" value = "object" />
    </position>
    <texcoord name = "_8" type = "vector2">
        <input name = "index" type = "integer" value = "0" />
    </texcoord>
    <multiply name = "_9" type = "vector2">
        <input name = "in1" type = "vector2" value = "1,-1" />
        <input name = "in2" type = "vector2" nodename = "_8" />
    </multiply>
    <add name = "_10" type = "vector2">
        <input name = "in1" type = "vector2" value = "0,1" />
        <input name = "in2" type = "vector2" nodename = "_9" />
    </add>
    <multiply name = "_11" type = "vector4">
        <input name = "in1" type = "vector4" nodename = "_56" />
        <input name = "in2" type = "vector4" value = "-1,-1,0,0" />
    </multiply>
    <UsdPrimvarReader name = "_12" type = "vector2">
        <input name = "varname" type = "string" value = "vertexUV1" />
    </UsdPrimvarReader>
    <multiply name = "_13" type = "vector2">
        <input name = "in1" type = "vector2" value = "1,-1" />
        <input name = "in2" type = "vector2" nodename = "_12" />
    </multiply>
    <add name = "_14" type = "vector2">
        <input name = "in1" type = "vector2" value = "0,1" />
        <input name = "in2" type = "vector2" nodename = "_13" />
    </add>
    <multiply name = "_15" type = "vector4">
        <input name = "in1" type = "vector4" nodename = "_60" />
        <input name = "in2" type = "vector4" value = "-1,-1,0,0" />
    </multiply>
    <multiply name = "_16" type = "vector3">
        <input name = "in1" type = "vector3" nodename = "_17" />
        <input name = "in2" type = "vector3" value = "1,1,-1" />
    </multiply>
    <normal name = "_17" type = "vector3">
        <input name = "space" type = "string" value = "object" />
    </normal>
    <dot name = "_18" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_19" />
    </dot>
    <RealityKitTexture2D name = "_19" type = "vector4">
        <input name = "file" type = "filename" nodename = "_20" />
        <input name = "texcoord" type = "vector2" nodename = "_21" />
        <input name = "min_filter" type = "string" value = "linear" />
        <input name = "mag_filter" type = "string" value = "linear" />
        <input name = "mip_filter" type = "string" value = "nearest" />
        <input name = "u_wrap_mode" type = "string" value = "repeat" />
        <input name = "v_wrap_mode" type = "string" value = "repeat" />
        <input name = "max_anisotropy" type = "integer" value = "1" />
    </RealityKitTexture2D>
    <dot name = "_20" type = "filename">
        <input name = "in" type = "filename" nodename = "_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D" />
    </dot>
    <constant name = "_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D" type = "filename">
        <input name = "value" type = "filename" value = "ERR" />
    </constant>
    <add name = "_21" type = "vector2">
        <input name = "in1" type = "vector2" nodename = "_22" />
        <input name = "in2" type = "vector2" nodename = "_25" />
    </add>
    <multiply name = "_22" type = "vector2">
        <input name = "in1" type = "vector2" nodename = "_23" />
        <input name = "in2" type = "vector2" nodename = "_24" />
    </multiply>
    <dot name = "_23" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_2" />
    </dot>
    <swizzle name = "_24" type = "vector2">
        <input name = "in" type = "vector4" nodename = "TextureTransform_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D" />
        <input name = "channels" type = "string" value = "xy" />
    </swizzle>
    <constant name = "TextureTransform_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D" type = "vector4">
        <input name = "value" type = "vector4" value = "1,1,0,0" />
    </constant>
    <swizzle name = "_25" type = "vector2">
        <input name = "in" type = "vector4" nodename = "TextureTransform_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D" />
        <input name = "channels" type = "string" value = "zw" />
    </swizzle>
    <dot name = "_26" type = "float">
        <input name = "in" type = "float" nodename = "_27" />
    </dot>
    <swizzle name = "_27" type = "float">
        <input name = "in" type = "vector4" nodename = "_19" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <dot name = "_28" type = "float">
        <input name = "in" type = "float" nodename = "_29" />
    </dot>
    <swizzle name = "_29" type = "float">
        <input name = "in" type = "vector4" nodename = "_19" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <dot name = "_30" type = "float">
        <input name = "in" type = "float" nodename = "_31" />
    </dot>
    <swizzle name = "_31" type = "float">
        <input name = "in" type = "vector4" nodename = "_19" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
    <dot name = "_32" type = "float">
        <input name = "in" type = "float" nodename = "_33" />
    </dot>
    <swizzle name = "_33" type = "float">
        <input name = "in" type = "vector4" nodename = "_19" />
        <input name = "channels" type = "string" value = "w" />
    </swizzle>
    <dot name = "_34" type = "vector4">
        <input name = "in" type = "vector4" nodename = "_35" />
    </dot>
    <RealityKitTexture2D name = "_35" type = "vector4">
        <input name = "file" type = "filename" nodename = "_36" />
        <input name = "texcoord" type = "vector2" nodename = "_37" />
        <input name = "min_filter" type = "string" value = "linear" />
        <input name = "mag_filter" type = "string" value = "linear" />
        <input name = "mip_filter" type = "string" value = "nearest" />
        <input name = "u_wrap_mode" type = "string" value = "repeat" />
        <input name = "v_wrap_mode" type = "string" value = "repeat" />
        <input name = "max_anisotropy" type = "integer" value = "1" />
    </RealityKitTexture2D>
    <dot name = "_36" type = "filename">
        <input name = "in" type = "filename" nodename = "_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D" />
    </dot>
    <constant name = "_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D" type = "filename">
        <input name = "value" type = "filename" value = "ERR" />
    </constant>
    <add name = "_37" type = "vector2">
        <input name = "in1" type = "vector2" nodename = "_38" />
        <input name = "in2" type = "vector2" nodename = "_41" />
    </add>
    <multiply name = "_38" type = "vector2">
        <input name = "in1" type = "vector2" nodename = "_39" />
        <input name = "in2" type = "vector2" nodename = "_40" />
    </multiply>
    <dot name = "_39" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_3" />
    </dot>
    <swizzle name = "_40" type = "vector2">
        <input name = "in" type = "vector4" nodename = "TextureTransform_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D" />
        <input name = "channels" type = "string" value = "xy" />
    </swizzle>
    <constant name = "TextureTransform_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D" type = "vector4">
        <input name = "value" type = "vector4" value = "1,1,0,0" />
    </constant>
    <swizzle name = "_41" type = "vector2">
        <input name = "in" type = "vector4" nodename = "TextureTransform_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D" />
        <input name = "channels" type = "string" value = "zw" />
    </swizzle>
    <dot name = "_42" type = "float">
        <input name = "in" type = "float" nodename = "_43" />
    </dot>
    <swizzle name = "_43" type = "float">
        <input name = "in" type = "vector4" nodename = "_35" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <dot name = "_44" type = "float">
        <input name = "in" type = "float" nodename = "_45" />
    </dot>
    <swizzle name = "_45" type = "float">
        <input name = "in" type = "vector4" nodename = "_35" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <dot name = "_46" type = "float">
        <input name = "in" type = "float" nodename = "_47" />
    </dot>
    <swizzle name = "_47" type = "float">
        <input name = "in" type = "vector4" nodename = "_35" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
    <dot name = "_48" type = "float">
        <input name = "in" type = "float" nodename = "_49" />
    </dot>
    <swizzle name = "_49" type = "float">
        <input name = "in" type = "vector4" nodename = "_35" />
        <input name = "channels" type = "string" value = "w" />
    </swizzle>
    <multiply name = "_50" type = "vector4">
        <input name = "in1" type = "vector4" nodename = "_63" />
        <input name = "in2" type = "vector4" nodename = "_18" />
    </multiply>
    <multiply name = "_51" type = "vector4">
        <input name = "in1" type = "vector4" nodename = "_34" />
        <input name = "in2" type = "vector4" nodename = "_4" />
    </multiply>
    <GeometryModification name = "_52" type = "vertexshader">
        <input name = "color" type = "color4" nodename = "_66" />
        <input name = "uv0" type = "vector2" nodename = "_67" />
        <input name = "uv1" type = "vector2" nodename = "_68" />
        <input name = "userAttribute" type = "vector4" nodename = "_70" />
    </GeometryModification>
    <realitykit_pbr name = "_53" type = "surfaceshader">
        <input name = "baseColor" type = "color3" nodename = "_64" />
        <input name = "metallic" type = "float" value = "0" />
        <input name = "emissiveColor" type = "color3" nodename = "_65" />
        <input name = "ambientOcclusion" type = "float" value = "1" />
        <input name = "specular" type = "float" value = "0.5" />
        <input name = "roughness" type = "float" nodename = "_54" />
        <input name = "clearcoat" type = "float" value = "0" />
        <input name = "clearcoatRoughness" type = "float" value = "0.01" />
    </realitykit_pbr>
    <surfacematerial name = "USD_CustomInterpolator" type = "material">
        <input name = "vertexshader" type = "vertexshader" nodename = "_52" />
        <input name = "surfaceshader" type = "surfaceshader" nodename = "_53" />
    </surfacematerial>
    <subtract name = "_54" type = "float">
        <input name = "in1" type = "float" value = "1" />
        <input name = "in2" type = "float" value = "0.5" />
    </subtract>
    <dot name = "_55" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_10" />
    </dot>
    <combine4 name = "_56" type = "vector4">
        <input name = "in1" type = "float" nodename = "_57" />
        <input name = "in2" type = "float" nodename = "_58" />
        <input name = "in3" type = "float" value = "0" />
        <input name = "in4" type = "float" value = "0" />
    </combine4>
    <swizzle name = "_57" type = "float">
        <input name = "in" type = "vector2" nodename = "_55" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_58" type = "float">
        <input name = "in" type = "vector2" nodename = "_55" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <dot name = "_59" type = "vector2">
        <input name = "in" type = "vector2" nodename = "_14" />
    </dot>
    <combine4 name = "_60" type = "vector4">
        <input name = "in1" type = "float" nodename = "_61" />
        <input name = "in2" type = "float" nodename = "_62" />
        <input name = "in3" type = "float" value = "0" />
        <input name = "in4" type = "float" value = "0" />
    </combine4>
    <swizzle name = "_61" type = "float">
        <input name = "in" type = "vector2" nodename = "_59" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_62" type = "float">
        <input name = "in" type = "vector2" nodename = "_59" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <swizzle name = "_63" type = "vector4">
        <input name = "in" type = "color4" nodename = "_1" />
        <input name = "channels" type = "string" value = "rgba" />
    </swizzle>
    <swizzle name = "_64" type = "color3">
        <input name = "in" type = "vector4" nodename = "_50" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
    <swizzle name = "_65" type = "color3">
        <input name = "in" type = "vector4" nodename = "_51" />
        <input name = "channels" type = "string" value = "xyz" />
    </swizzle>
    <swizzle name = "_66" type = "color4">
        <input name = "in" type = "vector4" nodename = "_5" />
        <input name = "channels" type = "string" value = "xyzw" />
    </swizzle>
    <swizzle name = "_67" type = "vector2">
        <input name = "in" type = "vector4" nodename = "_11" />
        <input name = "channels" type = "string" value = "xy" />
    </swizzle>
    <swizzle name = "_68" type = "vector2">
        <input name = "in" type = "vector4" nodename = "_15" />
        <input name = "channels" type = "string" value = "xy" />
    </swizzle>
    <dot name = "_69" type = "vector3">
        <input name = "in" type = "vector3" nodename = "_16" />
    </dot>
    <combine4 name = "_70" type = "vector4">
        <input name = "in1" type = "float" nodename = "_71" />
        <input name = "in2" type = "float" nodename = "_72" />
        <input name = "in3" type = "float" nodename = "_73" />
        <input name = "in4" type = "float" value = "0" />
    </combine4>
    <swizzle name = "_71" type = "float">
        <input name = "in" type = "vector3" nodename = "_69" />
        <input name = "channels" type = "string" value = "x" />
    </swizzle>
    <swizzle name = "_72" type = "float">
        <input name = "in" type = "vector3" nodename = "_69" />
        <input name = "channels" type = "string" value = "y" />
    </swizzle>
    <swizzle name = "_73" type = "float">
        <input name = "in" type = "vector3" nodename = "_69" />
        <input name = "channels" type = "string" value = "z" />
    </swizzle>
</materialx>
