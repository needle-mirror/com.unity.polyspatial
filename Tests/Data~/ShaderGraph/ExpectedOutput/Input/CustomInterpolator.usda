#usda 1.0
(
    metersPerUnit = 1
    upAxis = "Y"
)

def "MaterialX"
{
    def "Materials"
    {
        def Material "USD_CustomInterpolator"
        {
            asset inputs:_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            float4 inputs:TextureTransform_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D = (1, 1, 0, 0)
            asset inputs:_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D = @placeholder.png@ (colorSpace = "srgb_texture")
            float4 inputs:TextureTransform_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D = (1, 1, 0, 0)

            token outputs:mtlx:surface.connect = <realitykit_pbr.outputs:out>
            token outputs:realitykit:vertex.connect = <GeometryModifier.outputs:out>

            def Shader "realitykit_pbr"
            {
                uniform token info:id = "ND_realitykit_pbr_surfaceshader"
                color3f inputs:baseColor.connect = <../_64.outputs:out>
                float inputs:metallic = 0
                color3f inputs:emissiveColor.connect = <../_65.outputs:out>
                float inputs:ambientOcclusion = 1
                float inputs:specular = 0.5
                float inputs:roughness.connect = <../_54.outputs:out>
                float inputs:clearcoat = 0
                float inputs:clearcoatRoughness = 0.01
                token outputs:out
            }

            def Shader "GeometryModifier"
            {
                uniform token info:id = "ND_realitykit_geometrymodifier_vertexshader"
                color4f inputs:color.connect = <../_66.outputs:out>
                float2 inputs:uv0.connect = <../_67.outputs:out>
                float2 inputs:uv1.connect = <../_68.outputs:out>
                float4 inputs:userAttribute.connect = <../_70.outputs:out>
                token outputs:out
            }

            def Shader "_64"
            {
                uniform token info:id = "ND_swizzle_vector4_color3"
                float4 inputs:in.connect = <../_50.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_50"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_63.outputs:out>
                float4 inputs:in2.connect = <../_18.outputs:out>
                float4 outputs:out
            }

            def Shader "_63"
            {
                uniform token info:id = "ND_swizzle_color4_vector4"
                color4f inputs:in.connect = <../_1.outputs:out>
                string inputs:channels = "rgba"
                float4 outputs:out
            }

            def Shader "_1"
            {
                uniform token info:id = "ND_geomcolor_color4"
                color4f outputs:out
            }

            def Shader "_65"
            {
                uniform token info:id = "ND_swizzle_vector4_color3"
                float4 inputs:in.connect = <../_51.outputs:out>
                string inputs:channels = "xyz"
                color3f outputs:out
            }

            def Shader "_51"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_34.outputs:out>
                float4 inputs:in2.connect = <../_4.outputs:customAttribute>
                float4 outputs:out
            }

            def Shader "_4"
            {
                uniform token info:id = "ND_realitykit_surface_custom_attribute"
                float4 outputs:customAttribute
            }

            def Shader "_34"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_35.outputs:out>
                float4 outputs:out
            }

            def Shader "_35"
            {
                uniform token info:id = "ND_RealityKitTexture2D_vector4"
                asset inputs:file.connect = <../_36.outputs:out>
                float2 inputs:texcoord.connect = <../_37.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_37"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_38.outputs:out>
                float2 inputs:in2.connect = <../_41.outputs:out>
                float2 outputs:out
            }

            def Shader "_38"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_39.outputs:out>
                float2 inputs:in2.connect = <../_40.outputs:out>
                float2 outputs:out
            }

            def Shader "_39"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_3.outputs:out>
                float2 outputs:out
            }

            def Shader "_3"
            {
                uniform token info:id = "ND_UsdPrimvarReader_vector2"
                string inputs:varname = "vertexUV1"
                float2 outputs:out
            }

            def Shader "_36"
            {
                uniform token info:id = "ND_dot_filename"
                asset inputs:in.connect = <../.inputs:_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D>
                asset outputs:out
            }

            def Shader "_54"
            {
                uniform token info:id = "ND_subtract_float"
                float inputs:in1 = 1
                float inputs:in2 = 0.5
                float outputs:out
            }

            def Shader "_66"
            {
                uniform token info:id = "ND_swizzle_vector4_color4"
                float4 inputs:in.connect = <../_5.outputs:out>
                string inputs:channels = "xyzw"
                color4f outputs:out
            }

            def Shader "_5"
            {
                uniform token info:id = "ND_convert_vector3_vector4"
                float3 inputs:in.connect = <../_6.outputs:out>
                float4 outputs:out
            }

            def Shader "_6"
            {
                uniform token info:id = "ND_multiply_vector3"
                float3 inputs:in1.connect = <../_7.outputs:out>
                float3 inputs:in2 = (1, 1, -1)
                float3 outputs:out
            }

            def Shader "_7"
            {
                uniform token info:id = "ND_position_vector3"
                string inputs:space = "object"
                float3 outputs:out
            }

            def Shader "_67"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../_11.outputs:out>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_11"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_56.outputs:out>
                float4 inputs:in2 = (-1, -1, 0, 0)
                float4 outputs:out
            }

            def Shader "_56"
            {
                uniform token info:id = "ND_combine4_vector4"
                float inputs:in1.connect = <../_57.outputs:out>
                float inputs:in2.connect = <../_58.outputs:out>
                float inputs:in3 = 0
                float inputs:in4 = 0
                float4 outputs:out
            }

            def Shader "_58"
            {
                uniform token info:id = "ND_swizzle_vector2_float"
                float2 inputs:in.connect = <../_55.outputs:out>
                string inputs:channels = "y"
                float outputs:out
            }

            def Shader "_55"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_10.outputs:out>
                float2 outputs:out
            }

            def Shader "_10"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1 = (0, 1)
                float2 inputs:in2.connect = <../_9.outputs:out>
                float2 outputs:out
            }

            def Shader "_9"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1 = (1, -1)
                float2 inputs:in2.connect = <../_8.outputs:out>
                float2 outputs:out
            }

            def Shader "_8"
            {
                uniform token info:id = "ND_texcoord_vector2"
                int inputs:index = 0
                float2 outputs:out
            }

            def Shader "_57"
            {
                uniform token info:id = "ND_swizzle_vector2_float"
                float2 inputs:in.connect = <../_55.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_68"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../_15.outputs:out>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_15"
            {
                uniform token info:id = "ND_multiply_vector4"
                float4 inputs:in1.connect = <../_60.outputs:out>
                float4 inputs:in2 = (-1, -1, 0, 0)
                float4 outputs:out
            }

            def Shader "_60"
            {
                uniform token info:id = "ND_combine4_vector4"
                float inputs:in1.connect = <../_61.outputs:out>
                float inputs:in2.connect = <../_62.outputs:out>
                float inputs:in3 = 0
                float inputs:in4 = 0
                float4 outputs:out
            }

            def Shader "_62"
            {
                uniform token info:id = "ND_swizzle_vector2_float"
                float2 inputs:in.connect = <../_59.outputs:out>
                string inputs:channels = "y"
                float outputs:out
            }

            def Shader "_59"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_14.outputs:out>
                float2 outputs:out
            }

            def Shader "_14"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1 = (0, 1)
                float2 inputs:in2.connect = <../_13.outputs:out>
                float2 outputs:out
            }

            def Shader "_13"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1 = (1, -1)
                float2 inputs:in2.connect = <../_12.outputs:out>
                float2 outputs:out
            }

            def Shader "_12"
            {
                uniform token info:id = "ND_UsdPrimvarReader_vector2"
                string inputs:varname = "vertexUV1"
                float2 outputs:out
            }

            def Shader "_61"
            {
                uniform token info:id = "ND_swizzle_vector2_float"
                float2 inputs:in.connect = <../_59.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_70"
            {
                uniform token info:id = "ND_combine4_vector4"
                float inputs:in1.connect = <../_71.outputs:out>
                float inputs:in2.connect = <../_72.outputs:out>
                float inputs:in3.connect = <../_73.outputs:out>
                float inputs:in4 = 0
                float4 outputs:out
            }

            def Shader "_73"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_69.outputs:out>
                string inputs:channels = "z"
                float outputs:out
            }

            def Shader "_69"
            {
                uniform token info:id = "ND_dot_vector3"
                float3 inputs:in.connect = <../_16.outputs:out>
                float3 outputs:out
            }

            def Shader "_16"
            {
                uniform token info:id = "ND_multiply_vector3"
                float3 inputs:in1.connect = <../_17.outputs:out>
                float3 inputs:in2 = (1, 1, -1)
                float3 outputs:out
            }

            def Shader "_17"
            {
                uniform token info:id = "ND_normal_vector3"
                string inputs:space = "object"
                float3 outputs:out
            }

            def Shader "_72"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_69.outputs:out>
                string inputs:channels = "y"
                float outputs:out
            }

            def Shader "_71"
            {
                uniform token info:id = "ND_swizzle_vector3_float"
                float3 inputs:in.connect = <../_69.outputs:out>
                string inputs:channels = "x"
                float outputs:out
            }

            def Shader "_18"
            {
                uniform token info:id = "ND_dot_vector4"
                float4 inputs:in.connect = <../_19.outputs:out>
                float4 outputs:out
            }

            def Shader "_19"
            {
                uniform token info:id = "ND_RealityKitTexture2D_vector4"
                asset inputs:file.connect = <../_20.outputs:out>
                float2 inputs:texcoord.connect = <../_21.outputs:out>
                string inputs:min_filter = "linear"
                string inputs:mag_filter = "linear"
                string inputs:mip_filter = "nearest"
                string inputs:u_wrap_mode = "repeat"
                string inputs:v_wrap_mode = "repeat"
                int inputs:max_anisotropy = 1
                float4 outputs:out
            }

            def Shader "_21"
            {
                uniform token info:id = "ND_add_vector2"
                float2 inputs:in1.connect = <../_22.outputs:out>
                float2 inputs:in2.connect = <../_25.outputs:out>
                float2 outputs:out
            }

            def Shader "_25"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D>
                string inputs:channels = "zw"
                float2 outputs:out
            }

            def Shader "_22"
            {
                uniform token info:id = "ND_multiply_vector2"
                float2 inputs:in1.connect = <../_23.outputs:out>
                float2 inputs:in2.connect = <../_24.outputs:out>
                float2 outputs:out
            }

            def Shader "_24"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D>
                string inputs:channels = "xy"
                float2 outputs:out
            }

            def Shader "_23"
            {
                uniform token info:id = "ND_dot_vector2"
                float2 inputs:in.connect = <../_2.outputs:out>
                float2 outputs:out
            }

            def Shader "_2"
            {
                uniform token info:id = "ND_texcoord_vector2"
                int inputs:index = 0
                float2 outputs:out
            }

            def Shader "_20"
            {
                uniform token info:id = "ND_dot_filename"
                asset inputs:in.connect = <../.inputs:_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D>
                asset outputs:out
            }

            def Shader "_41"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D>
                string inputs:channels = "zw"
                float2 outputs:out
            }

            def Shader "_40"
            {
                uniform token info:id = "ND_swizzle_vector4_vector2"
                float4 inputs:in.connect = <../.inputs:TextureTransform_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D>
                string inputs:channels = "xy"
                float2 outputs:out
            }
        }
    }
}
