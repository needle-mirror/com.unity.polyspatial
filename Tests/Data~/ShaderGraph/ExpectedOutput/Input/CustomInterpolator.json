{"properties":[{"name":"_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D","type":6},{"name":"TextureTransform_SampleTexture2D_dde48ea7f0274d75aa3672cfec65d3b7_Texture_1_Texture2D","type":3},{"name":"_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D","type":6},{"name":"TextureTransform_SampleTexture2D_f702726f6d1743cf8a7973bf7d3a1068_Texture_1_Texture2D","type":3}]}